jina/types/arrays/mixins/traverse.py

```diff
@@ -18,16 +18,13 @@
 def _check_traversal_path_type(tp):
     is_deprecate_scheme = True
     if isinstance(tp, str):
-        tp = tp.split(',')
         is_deprecate_scheme = False
-
-    if not (
-        tp
-        and not isinstance(tp, str)
-        and isinstance(tp, Sequence)
-        and all(isinstance(p, str) for p in tp)
-    ):
-        raise TypeError('`traversal_paths` needs to be `Sequence[str]`')
+    elif isinstance(tp, Sequence) and all(isinstance(p, str) for p in tp):
```

Comment:
no need to check `is_deprecate_warning`. Just raise the `warning` here and that's it

Reply:
changed